File: ecos-records/src/main/java/ru/citeck/ecos/records3/record/op/atts/mixin/AttMixin.java
package ru.citeck.ecos.records3.record.op.atts.mixin;

import ru.citeck.ecos.records3.record.op.atts.schema.resolver.AttValueCtx;

import java.util.Collection;

public interface AttMixin {

    Object getAtt(String path, AttValueCtx value) throws Exception;

    Collection<String> getProvidedAtts();
}

File: ecos-records/src/main/java/ru/citeck/ecos/records3/record/mixin/AttMixin.kt
package ru.citeck.ecos.records3.record.mixin

import ru.citeck.ecos.records3.record.atts.value.AttValueCtx

interface AttMixin {

    @Throws(Exception::class)
    fun getAtt(path: String, value: AttValueCtx): Any?

    fun getProvidedAtts(): Collection<String>
}
